src/mon/PGMonitor.h

```diff
@@ -32,16 +32,15 @@ using namespace std;
 #include "msg/Messenger.h"
 #include "mon/MonitorDBStore.h"
 
-class MStatfs;
-class MMonCommand;
-class MGetPoolStats;
-
-class TextTable;
 class MPGStats;
+class PGStatService;
 
 class PGMonitor : public PaxosService {
-public:
   PGMap pg_map;
+  std::unique_ptr<PGStatService> pgservice;
+
+  bool do_delete = false;   ///< propose deleting pgmap data
+  bool did_delete = false;  ///< we already deleted pgmap data
```

Comment:
ah, we've got some haphazard did_delete checks in the PGMonitor. How about instead we expose that to the Monitor via an is_enabled() interface, and crash out if anybody tries to access a PaxosService which is not enabled?